packages/decode/src/decode.js

```diff
@@ -111,7 +117,39 @@ export const decode = async (
 export const decodeResponse = async (response, customDecoders = {}) => {
   let decoders = {...defaultDecoders, ...customDecoders}
 
-  const decodeInstructionsJson = response.encodedData
-
-  return await decode(decodeInstructionsJson, decoders)
+  switch(true) {
+    case(isGetTransactionStatus(response)):
+      return {
```

Comment:
should this return everything or just the things it decoded?

Reply:
could send lift the events out of the transaction and put them in the top level events array in the response?
If so could this then be done as detection if certain values are there, versus doing it based on a flag, which it currently does?

```
export const decodeResponse = async (response, customDecoders = {}) => {
  if (response.encodedData != null) return decode(response.encodedData, customDecoders)
  if (response.events != null) return decode(response.events, customDecoders)
  return null
}
```